### ISSUE TITLE
Music files deleted from server still appear in web interface - Need automatic synchronization

### ISSUE THREAD

**Issue by hackingchen** [navidrome/navidrome#3333]
### I confirm that:

- [X] I have searched the existing [open AND closed issues](https://github.com/navidrome/navidrome/issues?q=is%3Aissue) to see if an issue already exists for the bug I've encountered
- [X] I'm using the latest version (your issue may have been fixed already)

### Version

v0.53.3

### Current Behavior

Hello,

I've encountered an issue with the synchronization between the server's file system and the web interface. Here's the situation:

1. I've deleted music files directly from the server.
2. These deleted files still appear in the web interface's music list.
3. When I try to play these deleted files, it fails as expected.

Specifically:
- Is there a way to configure periodic synchronization between the server and the frontend?
- Can we set up an automatic process to remove entries from the database when the corresponding files are deleted?
- Are there any settings or configurations I should check to ensure this kind of periodic syncing between the file system and the database?
- If this feature doesn't exist, would it be possible to implement such a synchronization mechanism?

Any guidance on how to keep the web interface automatically and periodically updating to accurately reflect the current state of the music library would be greatly appreciated.

Thank you for your help and consideration!

### Expected Behavior

My concern is that I don't want to see these deleted files in the frontend anymore. What I'm hoping to achieve is:

- After files are deleted from the server, the frontend should automatically synchronize with the server at configurable intervals.
- This sync should retrieve the latest file information and update the frontend's database accordingly.

### Environment

```markdown
- OS:
- Browser:
- Client:
```

### How Navidrome is installed?

Docker

### Code of Conduct

- [X] I agree to follow Navidrome's Code of Conduct

**Comment by kgarner7** [comment]
See https://www.navidrome.org/docs/usage/configuration-options/#:~:text=false-,scanschedule,-ND_SCANSCHEDULE
The default schedule is 1 minute, and deleted files should be picked up after that. The quick rescan is also available from the activity button (next to user icon).
In the future, I believe there are plans to have filesystem watchers, which should detect changes almost immediately.

**Comment by jakemauer** [comment]
I'm running into an issue where I've deleted content from my music folder, re-run a quick scan and then the full system scan, and the deleted content remains in the web UI, even after refreshing and restarting Navidrome. I'm running version 
0.53.3 (13af8ed4) in docker.

**Comment by deluan** [comment]
@hackingchen and @jakemauer: Maybe the scanner in encountering an error and is not getting to the end of the process? Can you please post debug logs?

**Comment by jakemauer** [comment]
@deluan you're right, I was encountering an error with an album with special characters. This caused me to dig in and realize that my Proxmox setup where I'm mounting shares over CIFS was not using UTF8. Once I fixed that suddenly Navidrome was completely unblocked. Thank you!
